e2e/utils/test-utils.ts

```diff
@@ -1,9 +1,77 @@
-import {Page} from 'puppeteer';
+// import {performance} from "perf_hooks";
+import {JSHandle, Page} from 'puppeteer';
 import Checkbox from 'app/element/checkbox';
 import RadioButton from 'app/element/radiobutton';
 import TextOptions from 'app/element/text-options';
 import Textarea from 'app/element/textarea';
 import Textbox from 'app/element/textbox';
+import GoogleLoginPage from 'app/page/google-login';
+import HomePage from 'app/page/home-page';
+
+export async function signIn(page: Page) {
+  const loginPage = new GoogleLoginPage(page);
+  await loginPage.login();
+  // this element exists in DOM after user has logged in
+  await page.waitFor(() => document.querySelector('app-signed-in') !== null);
+  const homePage = new HomePage(page);
+  await homePage.waitForLoad();
+}
+
+/**
+ * <pre>
+ * Wait until spinner stops spinning, usually it indicate page is ready.
+ * </pre>
+ */
+export async function waitUntilNoSpinner(page: Page): Promise<void> {
```

Comment:
`waitUntilNoSpinner` method moved here from`base-page.ts`. It feels right to me to have it inside `test-utils.ts`.